webui/react/src/components/Router.tsx

```diff
@@ -12,13 +14,27 @@ interface Props {
 
 const Router: React.FC<Props> = (props: Props) => {
   const auth = Auth.useStateContext();
+  const checkAuth = useAuthCheck();
+  const setUI = UI.useActionContext();
+
+  useEffect(() => {
+    checkAuth();
+  }, [ checkAuth ]);
+
+  useEffect(() => {
+    if (auth.isAuthenticated) {
+      setUI({ type: UI.ActionType.HideSpinner });
+    }
+  }, [ auth.isAuthenticated, setUI ]);
 
   return (
     <Switch>
       {props.routes.map(config => {
         const { component, ...route } = config;
 
         if (route.needAuth && !auth.isAuthenticated) {
+          // Do not mount login page until auth is checked.
+          if (!auth.checked) return <Route key={route.id} {...route} />;
```

Comment:
This is sweet! So if it doesn't have `render` or `component` defined, it'll default to a blank screen?

Reply:
I initially went with an empty fragment but this worked. not necessarily a blank screen but no component for that portion of the app? I couldn't find a good reference to this in https://knowbody.github.io/react-router-docs/api/Route.html

the goal is to
- not render a component if needs auth and we're not authenticated
- avoid visual flickers in the UI